source/imaer-shared/src/test/java/nl/overheid/aerius/shared/emissions/ADMSRoadEmissionsCalculatorTest.java

```diff
@@ -0,0 +1,278 @@
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import nl.overheid.aerius.shared.domain.Substance;
+import nl.overheid.aerius.shared.domain.v2.base.TimeUnit;
+import nl.overheid.aerius.shared.domain.v2.geojson.Geometry;
+import nl.overheid.aerius.shared.domain.v2.source.ADMSRoadEmissionSource;
+import nl.overheid.aerius.shared.domain.v2.source.RoadEmissionSource;
+import nl.overheid.aerius.shared.domain.v2.source.road.CustomVehicles;
+import nl.overheid.aerius.shared.domain.v2.source.road.RoadStandardEmissionFactorsKey;
+import nl.overheid.aerius.shared.domain.v2.source.road.RoadStandardsInterpolationValues;
+import nl.overheid.aerius.shared.domain.v2.source.road.SpecificVehicles;
+import nl.overheid.aerius.shared.domain.v2.source.road.StandardVehicles;
+import nl.overheid.aerius.shared.domain.v2.source.road.ValuesPerVehicleType;
+import nl.overheid.aerius.shared.domain.v2.source.road.Vehicles;
+import nl.overheid.aerius.shared.exception.AeriusException;
+import nl.overheid.aerius.shared.geometry.GeometryCalculator;
+
+@ExtendWith(MockitoExtension.class)
+class ADMSRoadEmissionsCalculatorTest {
+
+  private static final String TEST_ROAD_AREA = "UK";
+  private static final String TEST_ROAD_TYPE_ADMS = "Lon1";
+
+  @Mock RoadEmissionFactorSupplier emissionFactorSupplier;
+  @Mock GeometryCalculator geometryCalculator;
+
+  ADMSRoadEmissionsCalculator emissionsCalculator;
+
+  @BeforeEach
+  void beforeEach() throws AeriusException {
+    emissionsCalculator = new ADMSRoadEmissionsCalculator(emissionFactorSupplier, geometryCalculator);
+  }
+
+  @Test
+  void testCalculateEmissionsADMS() throws AeriusException {
+    final ADMSRoadEmissionSource emissionSource = new ADMSRoadEmissionSource();
+    emissionSource.setRoadAreaCode(TEST_ROAD_AREA);
+    emissionSource.setRoadTypeCode(TEST_ROAD_TYPE_ADMS);
+    final Geometry geometry = mock(Geometry.class);
+    when(geometryCalculator.determineMeasure(geometry)).thenReturn(321.5);
+    final double gradient = 3.4;
+    emissionSource.setGradient(gradient);
+
+    // add 1 vehicles object of every type
+    emissionSource.getSubSources().add(createCustom());
+    emissionSource.getSubSources().add(createSpecific());
+    emissionSource.getSubSources().add(createStandard(gradient, true, true));
+
+    ensureSubSourceEmissionsEmpty(emissionSource);
+
+    final Map<Substance, Double> results = emissionsCalculator.calculateEmissions(emissionSource, geometry);
+
+    // Check total emissions
+    // 321.5 * (3703.7010 + 115.0685 + 673.9726 ) / (1000 * 1000)
+    assertEquals(1.4444, results.get(Substance.NOX), 1E-3);
```

Comment:
Preferable an assert should have a message text. Is it an idea to put the commented text above the asserts as part of such message?

Reply:
Did add messages to these unittests, but not the commented text above: those are more for checking how to arrive at the expected number (can chug em into Excel or something to see if these values match). I don't think they'll help to clarify anything if the tests fails.